File: backend/domain/model/puzzle.go
package model

type Puzzle struct {
	id       ID
	question string
	answer   string
}

func NewPuzzle(
	question string,
	answer string,
) (*Puzzle, error) {
	return &Puzzle{
		id:       NewID(),
		question: question,
		answer:   answer,
	}, nil
}

func (p Puzzle) ID() ID {
	return p.id
}

func (p Puzzle) Question() string {
	return p.question
}

func (p Puzzle) Answer() string {
	return p.answer
}

File: native/lib/infrastructure/repositoryimpl/mock/puzzle.dart
import 'package:lateral_thinking/domain/model/puzzle.dart';
import 'package:lateral_thinking/domain/repository/puzzle.dart';

class MockPuzzleRepository implements PuzzleRepository {
  @override
  Future<List<Puzzle>> listPuzzles() async {
    return [
      const Puzzle(id: 'test1', question: 'なぜカラスは鳴くのか？', answer: 'カラスは鳴くのではなく、鳴いていると聞こえるのだ'),
      const Puzzle(id: 'test2', question: 'なぜカラスは黒いのか？', answer: 'カラスは黒いのではなく、黒く見えるのだ'),
    ];
  }
}